pandas/core/groupby/ops.py

```diff
@@ -95,6 +96,64 @@
     get_indexer_dict,
 )
 
+_CYTHON_FUNCTIONS = {
+    "aggregate": {
+        "add": "group_add",
+        "prod": "group_prod",
+        "min": "group_min",
+        "max": "group_max",
+        "mean": "group_mean",
+        "median": "group_median",
+        "var": "group_var",
+        "first": "group_nth",
+        "last": "group_last",
+        "ohlc": "group_ohlc",
+    },
+    "transform": {
+        "cumprod": "group_cumprod",
+        "cumsum": "group_cumsum",
+        "cummin": "group_cummin",
+        "cummax": "group_cummax",
+        "rank": "group_rank",
+    },
+}
+
+
+@functools.lru_cache(maxsize=None)
+def _get_cython_function(kind: str, how: str, dtype: np.dtype, is_numeric: bool):
+
+    dtype_str = dtype.name
+    ftype = _CYTHON_FUNCTIONS[kind][how]
+
+    # see if there is a fused-type version of function
+    # only valid for numeric
+    f = getattr(libgroupby, ftype, None)
+    if f is not None and is_numeric:
+        return f
```

Comment:
any idea how many cases make it past here?  might be we can make some of this unnecessary on the cython side

Reply:
Quite some I think. Eg for sum, we have `group_add_..` which only has type-specific functions exposed (the same for mean, prod, var). Not directly sure why we expose some functions with `numeric` fused type, and others we expose the type specific ones (because those are also coded with fused types). That's a bigger change for another PR, though ;)

(now, an expensive part of this function is `dtype.name`, which is above this line anyway, so won't change that it's useful to cache this I think)